pkg/operator/controller/externaldns/pod.go

```diff
@@ -544,6 +540,23 @@ func (b *externalDNSVolumeBuilder) providerSpecificVolumes() []corev1.Volume {
 	return nil
 }
 
+// awsVolumes returns volumes needed for AWS provider
+func (b *externalDNSVolumeBuilder) awsVolumes() []corev1.Volume {
+	if len(b.secretName) == 0 {
+		return nil
+	}
+	return []corev1.Volume{
+		{
+			Name: awsCredentialsVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: b.secretName,
+				},
```

Comment:
Only `credentials` [key](https://github.com/openshift/cloud-credential-operator/blob/e31c1696e9e9e271bc53175e3bc0b52700dd8857/pkg/operator/constants/constants.go#L91-L93) needs to be consumed, similar to Azure or GCP volumes.

Reply:
There shouldn't be a credentials key. Basically the operator should copy over the credentials(in whatever form they may be) to the operand namespaced. And the operand should just consume that secret. I really don't see where the credentials key has to be used.

Reply:
I stand corrected. There is in fact a `credentials` keys in the secret provisioned by the CCO. But when the user provisions the credentials the secret does not have that field. So I believe what you mean is that the controller should handle both scenarios.

I would recommend that the controller which copies over the credentials from the `external-dns-operator` namespace to the `external-dns` namespace prepare the secret in the target namespace so that it is in the aws credentials file format. This way the deployment have to be generated in 2 different ways.

Reply:
Yes, the secret provided by the user may not have the field of for the credentials file. And yes, I [agree](https://github.com/openshift/external-dns-operator/pull/130#pullrequestreview-948257920) on the idea of taking care of `credentials` key in the credentials controller.